diff --git a/chrome/android/java/src/org/chromium/chrome/browser/ChromeSwitches.java b/chrome/android/java/src/org/chromium/chrome/browser/ChromeSwitches.java
index f756f83..35054ce 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/ChromeSwitches.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/ChromeSwitches.java
@@ -32,10 +32,6 @@ public abstract class ChromeSwitches {
      */
     public static final String AUTO_UPDATE_SERVER_CMD = "auto-update-server";
 
-    /** This will unblock the Chrome UI that is related to SignIn */
-    public static final String ENABLE_SUPPRESSED_CHROMIUM_FEATURES =
-            "enable-suppressed-chromium-features";
-
     /** Mimic a low end device */
     public static final String ENABLE_ACCESSIBILITY_TAB_SWITCHER =
             "enable-accessibility-tab-switcher";
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
index ddad560..1051547 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/FirstRunFlowSequencer.java
@@ -83,10 +83,6 @@ public abstract class FirstRunFlowSequencer  {
 
     @VisibleForTesting
     protected boolean isSyncAllowed() {
-        if (!CommandLine.getInstance().
-                hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES))
-            return false;
-
         return FeatureUtilities.canAllowSync(mActivity)
                 && !SigninManager.get(mActivity.getApplicationContext()).isSigninDisabledByPolicy();
     }
@@ -189,11 +185,8 @@ public abstract class FirstRunFlowSequencer  {
 
         freProperties.putBoolean(AccountFirstRunFragment.IS_CHILD_ACCOUNT, hasChildAccount);
 
-        if (CommandLine.getInstance().
-                hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES)) {
-            freProperties.putBoolean(FirstRunActivity.SHOW_DATA_REDUCTION_PAGE,
+        freProperties.putBoolean(FirstRunActivity.SHOW_DATA_REDUCTION_PAGE,
                     shouldShowDataReductionPage());
-        }
 
         onFlowIsKnown(freProperties);
         if (hasChildAccount || forceEduSignIn) {
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/ToSAndUMAFirstRunFragment.java b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/ToSAndUMAFirstRunFragment.java
index 3dc0c25..1c084c1 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/firstrun/ToSAndUMAFirstRunFragment.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/firstrun/ToSAndUMAFirstRunFragment.java
@@ -56,7 +56,6 @@ public class ToSAndUMAFirstRunFragment extends FirstRunPage {
             }
         });
 
-        if (CommandLine.getInstance().hasSwitch(ChromeSwitches.CRASH_LOG_SERVER_CMD)) {
             int paddingStart = getResources().getDimensionPixelSize(R.dimen.fre_tos_checkbox_padding);
             ApiCompatibilityUtils.setPaddingRelative(mSendReportCheckBox,
                     ApiCompatibilityUtils.getPaddingStart(mSendReportCheckBox) + paddingStart,
@@ -65,9 +64,6 @@ public class ToSAndUMAFirstRunFragment extends FirstRunPage {
                     mSendReportCheckBox.getPaddingBottom());
 
             mSendReportCheckBox.setChecked(!getPageDelegate().isNeverUploadCrashDump());
-        } else {
-            mSendReportCheckBox.setVisibility(View.GONE);
-        }
 
         mTosAndPrivacy.setMovementMethod(LinkMovementMethod.getInstance());
 
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/preferences/Preferences.java b/chrome/android/java/src/org/chromium/chrome/browser/preferences/Preferences.java
index 842cde5..854b167 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/preferences/Preferences.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/preferences/Preferences.java
@@ -214,13 +214,9 @@ public class Preferences extends AppCompatActivity implements
     public boolean onCreateOptionsMenu(Menu menu) {
         super.onCreateOptionsMenu(menu);
         // By default, every screen in Settings shows a "Help & feedback" menu item.
-        if(CommandLine.getInstance()
-                .hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES)) {
             MenuItem help = menu.add(
                     Menu.NONE, R.id.menu_id_general_help, Menu.CATEGORY_SECONDARY, R.string.menu_help);
             help.setIcon(R.drawable.ic_help_and_feedback);
-            return true;
-        }
         return true;
     }
 
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/preferences/datareduction/DataReductionPromoScreen.java b/chrome/android/java/src/org/chromium/chrome/browser/preferences/datareduction/DataReductionPromoScreen.java
index 9fc8ac9..32fc99b 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/preferences/datareduction/DataReductionPromoScreen.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/preferences/datareduction/DataReductionPromoScreen.java
@@ -43,10 +43,6 @@ public class DataReductionPromoScreen extends Dialog implements View.OnClickList
      * Launch the data reduction promo, if it needs to be displayed.
      */
     public static void launchDataReductionPromo(Activity parentActivity) {
-        if (!CommandLine.getInstance().
-                hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES)) {
-            return;
-        }
         // The promo is displayed if Chrome is launched directly (i.e., not with the intent to
         // navigate to and view a URL on startup), the instance is part of the field trial,
         // and the promo has not been displayed before.
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/preferences/privacy/PrivacyPreferences.java b/chrome/android/java/src/org/chromium/chrome/browser/preferences/privacy/PrivacyPreferences.java
index b596ebf..faa61e1 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/preferences/privacy/PrivacyPreferences.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/preferences/privacy/PrivacyPreferences.java
@@ -77,11 +77,7 @@ public class PrivacyPreferences extends PreferenceFragment
                 (ChromeBaseCheckBoxPreference) findPreference(PREF_CRASH_DUMP_UPLOAD_NO_CELLULAR);
 
         PreferenceScreen preferenceScreen = getPreferenceScreen();
-        if (!CommandLine.getInstance().hasSwitch(ChromeSwitches.CRASH_LOG_SERVER_CMD)) {
-            preferenceScreen.removePreference(uploadCrashDumpNoCellularPref);
-            preferenceScreen.removePreference(uploadCrashDumpPref);
-            preferenceScreen.removePreference(findPreference(PREF_USAGE_AND_CRASH_REPORTING));
-        } else if (privacyPrefManager.isCellularExperimentEnabled()) {
+        if (privacyPrefManager.isCellularExperimentEnabled()) {
             preferenceScreen.removePreference(uploadCrashDumpNoCellularPref);
             preferenceScreen.removePreference(uploadCrashDumpPref);
         } else {
@@ -287,12 +283,9 @@ public class PrivacyPreferences extends PreferenceFragment
     @Override
     public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
         menu.clear();
-        if(CommandLine.getInstance()
-                .hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES)) {
             MenuItem help = menu.add(
                     Menu.NONE, R.id.menu_id_targeted_help, Menu.NONE, R.string.menu_help);
             help.setIcon(R.drawable.ic_help_and_feedback);
-        }
     }
 
     @Override
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/preferences/website/LanguagePreferences.java b/chrome/android/java/src/org/chromium/chrome/browser/preferences/website/LanguagePreferences.java
index 6e65da9..3ad6556 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/preferences/website/LanguagePreferences.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/preferences/website/LanguagePreferences.java
@@ -80,13 +80,10 @@ public class LanguagePreferences extends PreferenceFragment {
     @Override
     public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
         menu.clear();
-        if(CommandLine.getInstance()
-                .hasSwitch(ChromeSwitches.ENABLE_SUPPRESSED_CHROMIUM_FEATURES)) {
             MenuItem help = menu.add(
                     Menu.NONE, R.id.menu_id_targeted_help, Menu.NONE, R.string.menu_help);
             help.setIcon(R.drawable.ic_help_and_feedback);
             help.setShowAsAction(MenuItem.SHOW_AS_ACTION_IF_ROOM);
-        }
 
         menu.add(Menu.NONE, R.id.menu_id_reset, Menu.NONE, R.string.reset_translate_defaults);
     }
