diff --git a/components/search_engines/prepopulated_engines.json b/components/search_engines/prepopulated_engines.json
index bc9769d..e6bf8d9 100644
--- a/components/search_engines/prepopulated_engines.json
+++ b/components/search_engines/prepopulated_engines.json
@@ -47,6 +47,17 @@
       "id": 1006
     },
 
+    "no_search": {
+      "name": "No Search",
+      "keyword": "nosearch",
+      "favicon_url": "about:blank",
+      "search_url": "http://{searchTerms}",
+      "suggest_url": "http://{searchTerms}",
+      "new_tab_url": "about:blank",
+      "type": "SEARCH_ENGINE_NOSEARCH",
+      "id": 7
+    },
+
     "aol": {
       "name": "AOL",
       "keyword": "aol.com",
diff --git a/components/search_engines/search_engine_type.h b/components/search_engines/search_engine_type.h
index 4e8ba58..fc1ad73 100644
--- a/components/search_engines/search_engine_type.h
+++ b/components/search_engines/search_engine_type.h
@@ -61,6 +61,7 @@ enum SearchEngineType {
   SEARCH_ENGINE_ZOZNAM,
   SEARCH_ENGINE_360,
   SEARCH_ENGINE_DUCKDUCKGO,
+  SEARCH_ENGINE_NOSEARCH,
   SEARCH_ENGINE_MAX          // Bounding value needed for UMA histogram macro.
 };
 
diff --git a/components/search_engines/template_url_prepopulate_data.cc b/components/search_engines/template_url_prepopulate_data.cc
index 9e57d09..53b1e11 100644
--- a/components/search_engines/template_url_prepopulate_data.cc
+++ b/components/search_engines/template_url_prepopulate_data.cc
@@ -50,441 +50,441 @@ namespace {
 
 // Default (for countries with no better engine set)
 const PrepopulatedEngine* engines_default[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // United Arab Emirates
 const PrepopulatedEngine* engines_AE[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Albania
 const PrepopulatedEngine* engines_AL[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Argentina
 const PrepopulatedEngine* engines_AR[] =
-    { &google, &duckduckgo, &bing, &yahoo_ar, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_ar, };
 
 // Austria
 const PrepopulatedEngine* engines_AT[] =
-    { &google, &duckduckgo, &bing, &yahoo_at, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_at, };
 
 // Australia
 const PrepopulatedEngine* engines_AU[] =
-    { &google, &duckduckgo, &bing, &yahoo_au, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_au, };
 
 // Bosnia and Herzegovina
 const PrepopulatedEngine* engines_BA[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Belgium
 const PrepopulatedEngine* engines_BE[] =
-    { &google, &duckduckgo, &bing, &yahoo, &yahoo_fr, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, &yahoo_fr, };
 
 // Bulgaria
 const PrepopulatedEngine* engines_BG[] =
-    { &google, &duckduckgo, &bing, &ask, };
+    { &google, &duckduckgo, &no_search, &bing, &ask, };
 
 // Bahrain
 const PrepopulatedEngine* engines_BH[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Burundi
 const PrepopulatedEngine* engines_BI[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Brunei
 const PrepopulatedEngine* engines_BN[] =
-    { &google, &duckduckgo, &yahoo_my, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_my, &bing, };
 
 // Bolivia
 const PrepopulatedEngine* engines_BO[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Brazil
 const PrepopulatedEngine* engines_BR[] =
-    { &google, &duckduckgo, &ask_br, &bing, &yahoo_br, };
+    { &google, &duckduckgo, &no_search, &ask_br, &bing, &yahoo_br, };
 
 // Belarus
 const PrepopulatedEngine* engines_BY[] =
-    { &google, &duckduckgo, &yandex_by, &mail_ru, };
+    { &google, &duckduckgo, &no_search, &yandex_by, &mail_ru, };
 
 // Belize
 const PrepopulatedEngine* engines_BZ[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Canada
 const PrepopulatedEngine* engines_CA[] =
-    { &google, &duckduckgo, &bing, &ask, &yahoo_ca, &yahoo_qc, };
+    { &google, &duckduckgo, &no_search, &bing, &ask, &yahoo_ca, &yahoo_qc, };
 
 // Switzerland
 const PrepopulatedEngine* engines_CH[] =
-    { &google, &duckduckgo, &bing, &yahoo_ch, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_ch, };
 
 // Chile
 const PrepopulatedEngine* engines_CL[] =
-    { &google, &duckduckgo, &bing, &yahoo_cl, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_cl, };
 
 // China
 const PrepopulatedEngine* engines_CN[] =
-    { &google, &duckduckgo, &baidu, &sogou, &so_360};
+    { &google, &duckduckgo, &no_search, &baidu, &sogou, &so_360};
 
 // Colombia
 const PrepopulatedEngine* engines_CO[] =
-    { &google, &duckduckgo, &bing, &yahoo_co, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_co, };
 
 // Costa Rica
 const PrepopulatedEngine* engines_CR[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Czech Republic
 const PrepopulatedEngine* engines_CZ[] =
-    { &google, &duckduckgo, &seznam, &bing, };
+    { &google, &duckduckgo, &no_search, &seznam, &bing, };
 
 // Germany
 const PrepopulatedEngine* engines_DE[] =
-    { &google, &duckduckgo, &bing, &yahoo_de };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_de };
 
 // Denmark
 const PrepopulatedEngine* engines_DK[] =
-    { &google, &duckduckgo, &bing, &yahoo_dk, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_dk, };
 
 // Dominican Republic
 const PrepopulatedEngine* engines_DO[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Algeria
 const PrepopulatedEngine* engines_DZ[] =
-    { &google, &duckduckgo, &bing, &yahoo_maktoob, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_maktoob, };
 
 // Ecuador
 const PrepopulatedEngine* engines_EC[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Estonia
 const PrepopulatedEngine* engines_EE[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Egypt
 const PrepopulatedEngine* engines_EG[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Spain
 const PrepopulatedEngine* engines_ES[] =
-    { &google, &duckduckgo, &bing, &yahoo_es, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_es, };
 
 // Faroe Islands
 const PrepopulatedEngine* engines_FO[] =
-    { &google, &duckduckgo, &bing, &ask, };
+    { &google, &duckduckgo, &no_search, &bing, &ask, };
 
 // Finland
 const PrepopulatedEngine* engines_FI[] =
-    { &google, &duckduckgo, &bing, &yahoo_fi, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_fi, };
 
 // France
 const PrepopulatedEngine* engines_FR[] =
-    { &google, &duckduckgo, &bing, &yahoo_fr, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_fr, };
 
 // United Kingdom
 const PrepopulatedEngine* engines_GB[] =
-    { &google, &duckduckgo, &bing, &yahoo_uk, &ask_uk, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_uk, &ask_uk, };
 
 // Greece
 const PrepopulatedEngine* engines_GR[] =
-    { &google, &duckduckgo, &bing, &yahoo_gr, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_gr, };
 
 // Guatemala
 const PrepopulatedEngine* engines_GT[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Hong Kong
 const PrepopulatedEngine* engines_HK[] =
-    { &google, &duckduckgo, &yahoo_hk, &baidu, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_hk, &baidu, &bing, };
 
 // Honduras
 const PrepopulatedEngine* engines_HN[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Croatia
 const PrepopulatedEngine* engines_HR[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Hungary
 const PrepopulatedEngine* engines_HU[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Indonesia
 const PrepopulatedEngine* engines_ID[] =
-    { &google, &duckduckgo, &yahoo_id, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_id, &bing, };
 
 // Ireland
 const PrepopulatedEngine* engines_IE[] =
-    { &google, &duckduckgo, &bing, &yahoo_uk, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_uk, };
 
 // Israel
 const PrepopulatedEngine* engines_IL[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // India
 const PrepopulatedEngine* engines_IN[] =
-    { &google, &duckduckgo, &bing, &yahoo_in, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_in, };
 
 // Iraq
 const PrepopulatedEngine* engines_IQ[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Iran
 const PrepopulatedEngine* engines_IR[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Iceland
 const PrepopulatedEngine* engines_IS[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Italy
 const PrepopulatedEngine* engines_IT[] =
-    { &google, &duckduckgo, &virgilio, &bing, };
+    { &google, &duckduckgo, &no_search, &virgilio, &bing, };
 
 // Jamaica
 const PrepopulatedEngine* engines_JM[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Jordan
 const PrepopulatedEngine* engines_JO[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Japan
 const PrepopulatedEngine* engines_JP[] =
-    { &google, &duckduckgo, &yahoo_jp, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_jp, &bing, };
 
 // Kenya
 const PrepopulatedEngine* engines_KE[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Kuwait
 const PrepopulatedEngine* engines_KW[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // South Korea
 const PrepopulatedEngine* engines_KR[] =
-    { &google, &duckduckgo, &naver, &daum, };
+    { &google, &duckduckgo, &no_search, &naver, &daum, };
 
 // Kazakhstan
 const PrepopulatedEngine* engines_KZ[] =
-    { &google, &duckduckgo, &mail_ru, &yandex_kz, };
+    { &google, &duckduckgo, &no_search, &mail_ru, &yandex_kz, };
 
 // Lebanon
 const PrepopulatedEngine* engines_LB[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Liechtenstein
 const PrepopulatedEngine* engines_LI[] =
-    { &google, &duckduckgo, &bing, &yahoo_de, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_de, };
 
 // Lithuania
 const PrepopulatedEngine* engines_LT[] =
-    { &google, &duckduckgo, &bing, &yandex_ru, };
+    { &google, &duckduckgo, &no_search, &bing, &yandex_ru, };
 
 // Luxembourg
 const PrepopulatedEngine* engines_LU[] =
-    { &google, &duckduckgo, &bing, &yahoo_fr, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_fr, };
 
 // Latvia
 const PrepopulatedEngine* engines_LV[] =
-    { &google, &duckduckgo, &yandex_ru, &bing, };
+    { &google, &duckduckgo, &no_search, &yandex_ru, &bing, };
 
 // Libya
 const PrepopulatedEngine* engines_LY[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Morocco
 const PrepopulatedEngine* engines_MA[] =
-    { &google, &duckduckgo, &bing, &yahoo_maktoob, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_maktoob, };
 
 // Monaco
 const PrepopulatedEngine* engines_MC[] =
-    { &google, &duckduckgo, &yahoo_fr, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_fr, &bing, };
 
 // Moldova
 const PrepopulatedEngine* engines_MD[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Montenegro
 const PrepopulatedEngine* engines_ME[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Macedonia
 const PrepopulatedEngine* engines_MK[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Mexico
 const PrepopulatedEngine* engines_MX[] =
-    { &google, &duckduckgo, &bing, &yahoo_mx, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_mx, };
 
 // Malaysia
 const PrepopulatedEngine* engines_MY[] =
-    { &google, &duckduckgo, &yahoo_my, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_my, &bing, };
 
 // Nicaragua
 const PrepopulatedEngine* engines_NI[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Netherlands
 const PrepopulatedEngine* engines_NL[] =
-    { &google, &duckduckgo, &yahoo_nl, &vinden, };
+    { &google, &duckduckgo, &no_search, &yahoo_nl, &vinden, };
 
 // Norway
 const PrepopulatedEngine* engines_NO[] =
-    { &google, &duckduckgo, &bing, &kvasir, };
+    { &google, &duckduckgo, &no_search, &bing, &kvasir, };
 
 // New Zealand
 const PrepopulatedEngine* engines_NZ[] =
-    { &google, &duckduckgo, &bing, &yahoo_nz, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_nz, };
 
 // Oman
 const PrepopulatedEngine* engines_OM[] =
-    { &google, &duckduckgo, &bing, &yahoo_maktoob, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_maktoob, };
 
 // Panama
 const PrepopulatedEngine* engines_PA[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Peru
 const PrepopulatedEngine* engines_PE[] =
-    { &google, &duckduckgo, &bing, &yahoo_pe, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_pe, };
 
 // Philippines
 const PrepopulatedEngine* engines_PH[] =
-    { &google, &duckduckgo, &yahoo_ph, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_ph, &bing, };
 
 // Pakistan
 const PrepopulatedEngine* engines_PK[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Puerto Rico
 const PrepopulatedEngine* engines_PR[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Poland
 const PrepopulatedEngine* engines_PL[] =
-    { &google, &duckduckgo, &onet, &bing, };
+    { &google, &duckduckgo, &no_search, &onet, &bing, };
 
 // Portugal
 const PrepopulatedEngine* engines_PT[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Paraguay
 const PrepopulatedEngine* engines_PY[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Qatar
 const PrepopulatedEngine* engines_QA[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Romania
 const PrepopulatedEngine* engines_RO[] =
-    { &google, &duckduckgo, &yahoo_ro, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_ro, &bing, };
 
 // Serbia
 const PrepopulatedEngine* engines_RS[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Russia
 const PrepopulatedEngine* engines_RU[] =
-    { &google, &duckduckgo, &yandex_ru, &mail_ru, };
+    { &google, &duckduckgo, &no_search, &yandex_ru, &mail_ru, };
 
 // Rwanda
 const PrepopulatedEngine* engines_RW[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Saudi Arabia
 const PrepopulatedEngine* engines_SA[] =
-    { &google, &duckduckgo, &yahoo_maktoob, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_maktoob, &bing, };
 
 // Sweden
 const PrepopulatedEngine* engines_SE[] =
-    { &google, &duckduckgo, &bing, &yahoo_se, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_se, };
 
 // Singapore
 const PrepopulatedEngine* engines_SG[] =
-    { &google, &duckduckgo, &yahoo_sg, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_sg, &bing, };
 
 // Slovenia
 const PrepopulatedEngine* engines_SI[] =
-    { &google, &duckduckgo, &najdi, &ask, };
+    { &google, &duckduckgo, &no_search, &najdi, &ask, };
 
 // Slovakia
 const PrepopulatedEngine* engines_SK[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // El Salvador
 const PrepopulatedEngine* engines_SV[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Syria
 const PrepopulatedEngine* engines_SY[] =
-    { &google, &duckduckgo, &bing, &yahoo_maktoob, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_maktoob, };
 
 // Thailand
 const PrepopulatedEngine* engines_TH[] =
-    { &google, &duckduckgo, &yahoo_th, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_th, &bing, };
 
 // Tunisia
 const PrepopulatedEngine* engines_TN[] =
-    { &google, &duckduckgo, &bing, &yahoo_maktoob, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_maktoob, };
 
 // Turkey
 const PrepopulatedEngine* engines_TR[] =
-    { &google, &duckduckgo, &bing, &yahoo_tr, &yandex_tr, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_tr, &yandex_tr, };
 
 // Trinidad and Tobago
 const PrepopulatedEngine* engines_TT[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Taiwan
 const PrepopulatedEngine* engines_TW[] =
-    { &google, &duckduckgo, &yahoo_tw, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_tw, &bing, };
 
 // Tanzania
 const PrepopulatedEngine* engines_TZ[] =
-    { &google, &duckduckgo, &yahoo, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo, &bing, };
 
 // Ukraine
 const PrepopulatedEngine* engines_UA[] =
-    { &google, &duckduckgo, &yandex_ua, &bing, };
+    { &google, &duckduckgo, &no_search, &yandex_ua, &bing, };
 
 // United States
 const PrepopulatedEngine* engines_US[] =
-    { &google, &duckduckgo, &bing, &yahoo, &aol, &ask, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, &aol, &ask, };
 
 // Uruguay
 const PrepopulatedEngine* engines_UY[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Venezuela
 const PrepopulatedEngine* engines_VE[] =
-    { &google, &duckduckgo, &bing, &yahoo_ve, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_ve, };
 
 // Vietnam
 const PrepopulatedEngine* engines_VN[] =
-    { &google, &duckduckgo, &yahoo_vn, &bing, };
+    { &google, &duckduckgo, &no_search, &yahoo_vn, &bing, };
 
 // Yemen
 const PrepopulatedEngine* engines_YE[] =
-    { &google, &duckduckgo, &bing, &yahoo_maktoob, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo_maktoob, };
 
 // South Africa
 const PrepopulatedEngine* engines_ZA[] =
-    { &google, &duckduckgo, &bing, &yahoo, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, };
 
 // Zimbabwe
 const PrepopulatedEngine* engines_ZW[] =
-    { &google, &duckduckgo, &bing, &yahoo, &ask, };
+    { &google, &duckduckgo, &no_search, &bing, &yahoo, &ask, };
 
 // A list of all the engines that we know about.
 const PrepopulatedEngine* kAllEngines[] = {
   // Prepopulated engines:
   &aol,          &ask,          &ask_br,       &ask_uk,       &baidu,
-  &bing,         &daum,&duckduckgo,&google,    &kvasir,       &mail_ru,
+  &bing,         &daum,&duckduckgo,&google,&no_search,&kvasir,&mail_ru,
   &najdi,        &naver,        &onet,         &seznam,       &sogou,
   &vinden,       &virgilio,     &yahoo,        &yahoo_ar,     &yahoo_at,
   &yahoo_au,     &yahoo_br,     &yahoo_ca,     &yahoo_ch,     &yahoo_cl,
